lib/BackgroundJob/Tasks/CheckRequirementsTask.php

```diff
@@ -64,27 +73,28 @@ public function description() {
 	public function execute(FaceRecognitionContext $context) {
 		$this->setContext($context);
 
-		$model = $this->settingsService->getCurrentFaceModel();
+		$modelId = $this->settingsService->getCurrentFaceModel();
 
-		$req = new Requirements($context->modelService, $model);
-
-		if (!$req->pdlibLoaded()) {
-			$error_message = "PDLib is not loaded. Cannot continue";
+		$model = $this->modelManager->getModel($modelId);
+		if (!$model) {
+			$error_message =
+				"Seems that you don't have any model installed\n" .
+				"Please contact administrator to change models you are using for face recognition\n" .
+				"or reinstall them with the 'occ face:setup --model' command. \n\n" .
+				"Fill an issue here if that doesn't help: https://github.com/matiasdelellis/facerecognition/issues";
 			$this->logInfo($error_message);
 			return false;
 		}
 
-		if (!$req->modelFilesPresent()) {
-			$error_message =
-				"Files of model with ID " . $model . " are not present in models/ directory.\n" .
-				"Please contact administrator to change models you are using for face recognition\n" .
-				"or reinstall them with the 'occ face:setup' command. \n" .
-				"Fill an issue here if that doesn't help: https://github.com/matiasdelellis/facerecognition/issues";
+		if (!$model->meetDependencies()) {
+			$error_message = "Seems that you don't meet the dependencies to use the model " . $modelId .": " . $model->getName();
+			// Document models on wiki and print link here.
 			$this->logInfo($error_message);
 			return false;
 		}
 
-		if (!$req->hasEnoughMemory()) {
+		$systemMemory = MemoryLimits::getSystemMemory();
```

Comment:
hm, just a thought....maybe you can abstract away memory requirements under `meetDependecies`? Memory requirements are specific to model, right? HOG might not have memory limit to be checked, or some other model (imagine remote service as a model) will not have it. I would argue that memory requirements is just another requirements that is specific to model

Reply:
> Memory requirements are specific to model, right?.

Is much more important the memory usage of model, but it still consumes a lot of memory when cluster the faces.. Do you remember your OOM when grouping with 0.6 sensitivity?

My idea was remove Requirements, but maybe I should keep it to leave this class cleaner.